test/e2e/daemon_set.go

```diff
@@ -263,6 +264,53 @@ var _ = framework.KubeDescribe("Daemon set [Serial]", func() {
 		Expect(c.DaemonSets(ns).Delete(dsName)).NotTo(HaveOccurred())
 
 	})
+
+	It("should update pod when spec was updated and update strategy is rolling update", func() {
```

Comment:
You also need a test for paused daemon sets, for rollbacks, and for the noop strategy. I would be fine to have them in a follow-up.